### PR TITLE
Vermont Elderly or Permanently Totally Disabled Tax Credit 

### LINKED ISSUE #3281: Vermont Elderly or Permanently Totally Disabled Tax Credit
Reference: 
https://www.law.cornell.edu/uscode/text/26/22
https://tax.vermont.gov/individuals/personal-income-tax/tax-credits

### PR DESCRIPTION
**Pull request by Shinuing**
Fixes #3281

### Summary
🆕📝🔧

This pull request adds the Vermont elderly or disabled tax credit feature to the policy engine, along with a test case and a changelog entry. It also resolves merge conflicts in the Utah retirement credit test file.

> _We are the coders of the tax reform_
> _We add the features that the people demand_
> _We write the parameters, variables, and tests_
> _We fix the conflicts with our metal skills_

### Walkthrough
*  Add Vermont elderly or disabled tax credit feature ([link](https://github.com/PolicyEngine/policyengine-us/pull/3282/files?diff=unified&w=0#diff-5885042762833a4aef51405a4036067e4e6aefa7a5091a2987d1d9d1fc2647e8R1-R4), [link](https://github.com/PolicyEngine/policyengine-us/pull/3282/files?diff=unified&w=0#diff-7750d4f4a2ab07333d5b528a4cb66652bb9aed1652e369bae8df930bb79b3232R1-R14), [link](https://github.com/PolicyEngine/policyengine-us/pull/3282/files?diff=unified&w=0#diff-57d70e1240fa12d13add69a28f46cb859cc90841d6a7acbe57cba28b1cd416f3R1-R10), [link](https://github.com/PolicyEngine/policyengine-us/pull/3282/files?diff=unified&w=0#diff-75fab679a4c34883c948565e4cee0a036b97210eb600c1ea4f430ec1269e9e1eR1-R7), [link](https://github.com/PolicyEngine/policyengine-us/pull/3282/files?diff=unified&w=0#diff-0aa43545aa8c47a909789cd8b31b59a8c94073496b3329220a988e544aacaeecR1-R20), [link](https://github.com/PolicyEngine/policyengine-us/pull/3282/files?diff=unified&w=0#diff-31c4776bb55b53bd14a6d9110b77dfb91d0598d870fc238ef2e9c7c088629279R11-R12))
* Remove merge conflict markers from `ut_retirement_credit_max.yaml` test case file ([link](https://github.com/PolicyEngine/policyengine-us/pull/3282/files?diff=unified&w=0#diff-3e708e6f690ef51f3c0101b558032ed7a73a4e3fbca556b21802e10be014cb49R119-R120), [link](https://github.com/PolicyEngine/policyengine-us/pull/3282/files?diff=unified&w=0#diff-3e708e6f690ef51f3c0101b558032ed7a73a4e3fbca556b21802e10be014cb49R143))

### CODE REVIEW

File: policyengine_us/parameters/gov/states/vt/tax/income/agi/exclusions/elderly_or_disabled/eligibility/age.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+description: Vermont provides elderly or permanently and totally disabled tax credit to filters at or above this age.
+values:
+  2022-01-01: 65
+metadata:
+  unit: year
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/vt/tax/income/agi/exclusions/elderly_or_disabled/eligibility/age.yaml:5]:
```suggestion
  unit: year
  period: year
```

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/vt/tax/income/agi/exclusions/elderly_or_disabled/eligibility/age.yaml:5]: for all parameter that are missing it

File: policyengine_us/parameters/gov/states/vt/tax/income/agi/exclusions/elderly_or_disabled/eligibility/nontaxable_income_limit.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/vt/tax/income/agi/exclusions/elderly_or_disabled/eligibility/nontaxable_income_limit.yaml:1]: rename to income_limit.yaml

File: policyengine_us/parameters/gov/states/vt/tax/income/agi/exclusions/elderly_or_disabled/eligibility/nontaxable_income_limit.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+description: Vermont limits its elderly or permanently and totally disabled tax credit to filers with other nontaxable income below this amount, based on filing status.
+single:
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/vt/tax/income/agi/exclusions/elderly_or_disabled/eligibility/nontaxable_income_limit.yaml:2]:
```suggestion
SINGLE:
```

File: policyengine_us/parameters/gov/states/vt/tax/income/agi/exclusions/elderly_or_disabled/eligibility/nontaxable_income_limit.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  values:
+    2022-01-01: 5_000
+separate:
+  values:
+    2022-01-01: 3_750
+joint_one_qualified:
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/vt/tax/income/agi/exclusions/elderly_or_disabled/eligibility/nontaxable_income_limit.yaml:8]:
```suggestion
JOINT:
```

File: policyengine_us/parameters/gov/states/vt/tax/income/agi/exclusions/elderly_or_disabled/eligibility/agi_limit.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/vt/tax/income/agi/exclusions/elderly_or_disabled/eligibility/agi_limit.yaml:1]: we dont need this variable 
- we can just multiply the `elderly_disabled_credit` with the rate

File: policyengine_us/variables/gov/states/vt/tax/income/adjusted_gross_income/subtractions/vt_elderly_or_disabled_exclusion/vt_elderly_or_disabled_credit_exclusion.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    value_type = float
+    entity = TaxUnit
+    definition_period = YEAR
+    label = "Vermont Elderly or disabled credit"
+    documentation = "Schedule R credit for the elderly and the disabled"
+    unit = USD
+    reference = (
+        "https://tax.vermont.gov/individuals/personal-income-tax/tax-credits"
+    )
+    defined_for = "vt_elderly_or_disabled_credit_exclusion_eligible"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/vt/tax/income/adjusted_gross_income/subtractions/vt_elderly_or_disabled_exclusion/vt_elderly_or_disabled_credit_exclusion.py:14]:
```suggestion
    defined_for = StateCode.VT
```

File: policyengine_us/parameters/gov/states/vt/tax/income/agi/exclusions/elderly_or_disabled/match.yaml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+description: Vermont matches this fraction of the federal credit for elderly or the disabled. 
+values:
+  2022-01-01: 0.24
+metadata:
+  unit: /1
+  period: year
+  label: Vermont elderly or the disabled tax credit match
+  reference:
+    - title: Vermont - Department of Taxes - Tax Credits and Adjustments for Individuals - Elderly or Permanently Totally Disabled Tax Credit
+      href: https://tax.vermont.gov/individuals/personal-income-tax/tax-credits
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/vt/tax/income/agi/exclusions/elderly_or_disabled/match.yaml:10]: need 2022 and 2021 tax form reference

File: policyengine_us/variables/gov/states/vt/tax/income/adjusted_gross_income/subtractions/vt_elderly_or_disabled_exclusion/vt_elderly_or_disabled_credit_exclusion.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/vt/tax/income/adjusted_gross_income/subtractions/vt_elderly_or_disabled_exclusion/vt_elderly_or_disabled_credit_exclusion.py:1]: we need to create a non_refunable.yaml file and an adds function in the vt_non_refundable_credits (double check if it non refundable)

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/vt/tax/income/adjusted_gross_income/subtractions/vt_elderly_or_disabled_exclusion/vt_elderly_or_disabled_credit_exclusion.py:1]: also move structure to credits folder -> same for parameters and tests

File: policyengine_us/parameters/gov/states/vt/tax/income/credits/elderly_or_disabled/match.yaml [hunk 6]
```diff
@@ -0,0 +1,14 @@
+description: Vermont matches this fraction of the federal credit for elderly or the disabled. 
+values:
+  2021-01-01: 0.24
+metadata:
+  unit: /1
+  period: year
+  label: Vermont elderly or the disabled tax credit match
+  reference:
+    - title: 32 V.S.A. §5822. Tax on income of individuals, estates, and trusts (d)(1)
+      href: https://legislature.vermont.gov/statutes/section/32/151/05828b
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/vt/tax/income/credits/elderly_or_disabled/match.yaml:10]: I am not sure if this link has a d 1 part

File: policyengine_us/parameters/gov/states/vt/tax/income/credits/non_refundable.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/vt/tax/income/credits/non_refundable.yaml:1]: dont need this as we are going to make it a subtraction

File: policyengine_us/tests/policy/baseline/gov/states/ut/tax/income/credits/retirement_credit/ut_retirement_credit_max.yaml [hunk 7]
```diff
@@ -116,6 +116,8 @@
         state_code: UT
   output:  
     ut_retirement_credit_max: 662.5
+<<<<<<< HEAD
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/ut/tax/income/credits/retirement_credit/ut_retirement_credit_max.yaml:119]: delete

File: policyengine_us/tests/policy/baseline/gov/states/ut/tax/income/credits/retirement_credit/ut_retirement_credit_max.yaml [hunk 8]
```diff
@@ -138,3 +140,4 @@
         state_code: UT
   output: 
     ut_retirement_credit_max: 300
+>>>>>>> 83fde8de662f071e496f0c3d40c23228584362e1
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/ut/tax/income/credits/retirement_credit/ut_retirement_credit_max.yaml:143]: delete

File: policyengine_us/tests/policy/baseline/gov/states/ut/tax/income/credits/retirement_credit/ut_retirement_credit_max.yaml (hunk 7 above)
Comment (PavelMakarchuk) [reply policyengine_us/tests/policy/baseline/gov/states/ut/tax/income/credits/retirement_credit/ut_retirement_credit_max.yaml:119]: probably a git pull upstream master

File: policyengine_us/variables/gov/states/vt/tax/income/vt_non_refundable_credits.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/vt/tax/income/vt_non_refundable_credits.py:1]: dont need this as we add to subtractions

File: policyengine_us/variables/gov/states/vt/tax/income/credits/elderly_or_disabled/vt_elderly_or_disabled_credit.py [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+    definition_period = YEAR
+    label = "Vermont Elderly or disabled credit"
+    documentation = "Schedule R credit for the elderly and the disabled"
+    unit = USD
+    reference = (
+        "https://tax.vermont.gov/individuals/personal-income-tax/tax-credits"
+    )
+    defined_for = StateCode.VT
+
+    def formula(tax_unit, period, parameters):
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/vt/tax/income/credits/elderly_or_disabled/vt_elderly_or_disabled_credit.py:16]: could you add a comment saying that the Investment Tax Credit and Vermont Farm Income Averaging Credit also get computed but leave them out here

File: policyengine_us/variables/gov/states/vt/tax/income/credits/elderly_or_disabled/vt_elderly_or_disabled_credit.py [hunk 10]
```diff
@@ -0,0 +1,20 @@
+from policyengine_us.model_api import *
+
+
+class vt_elderly_or_disabled_credit(Variable):
+    value_type = float
+    entity = TaxUnit
+    definition_period = YEAR
+    label = "Vermont Elderly or disabled credit"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/vt/tax/income/credits/elderly_or_disabled/vt_elderly_or_disabled_credit.py:8]:
```suggestion
    label = "Vermont elderly or disabled credit"
```

File: policyengine_us/parameters/gov/states/vt/tax/income/credits/elderly_or_disabled/match.yaml (hunk 6 above)
Comment (Shinuing) [reply policyengine_us/parameters/gov/states/vt/tax/income/credits/elderly_or_disabled/match.yaml:10]: Updated

File: policyengine_us/variables/gov/states/vt/tax/income/adjusted_gross_income/subtractions/elderly_or_disabled/vt_elderly_or_disabled_credit.py [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+    entity = TaxUnit
+    definition_period = YEAR
+    label = "Vermont elderly or disabled credit"
+    documentation = "Schedule R credit for the elderly and the disabled"
+    unit = USD
+    reference = (
+        "https://tax.vermont.gov/individuals/personal-income-tax/tax-credits"
+    )
+    defined_for = StateCode.VT
+    # the Investment Tax Credit and Vermont Farm Income Averaging Credit also get computed but leave them out here
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/vt/tax/income/adjusted_gross_income/subtractions/elderly_or_disabled/vt_elderly_or_disabled_credit.py:15]:
```suggestion
    # The Investment Tax Credit and Vermont Farm Income Averaging Credit are also subject to the match
    # these are currently not included 
```

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_us/variables/gov/states/vt/tax/income/credits/vt_elderly_or_disabled_credit.py [hunk 12]
```diff
@@ -16,7 +16,7 @@ class vt_elderly_or_disabled_credit(Variable):
     # these are currently not included
 
     def formula(tax_unit, period, parameters):
-        p = parameters(
-            period
-        ).gov.states.vt.tax.income.agi.exclusions.elderly_or_disabled
-        return p.match * tax_unit("elderly_disabled_credit", period)
+        p = parameters(period).gov.states.vt.tax.income.credits
+        return p.elderly_or_disabled * tax_unit(
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/vt/tax/income/credits/vt_elderly_or_disabled_credit.py:20]:
```suggestion
        us_elderly_disabled_credit = tax_unit("elderly_disabled_credit", period)
        return p.elderly_or_disabled * us_elderly_disabled_credit
```

**Review by MaxGhenis — APPROVED** [review]
good to merge after this nit for readability

**Pull request merged**